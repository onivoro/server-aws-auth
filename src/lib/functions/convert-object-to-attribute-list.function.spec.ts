import { convertObjectToAttributeList } from './convert-object-to-attribute-list.function';

describe('convertObjectToAttributeList', () => {
    it.each([
        null, undefined, '', 0
    ])('returns empty array for falsey values like => %j', (input) => {
        expect(convertObjectToAttributeList(input as any)).toEqual([]);
    });

    it.each([
        { email: null, phone: undefined, address: '', number: 0, age: 337 },
    ])('formats object to array, given %j', (input) => {
        expect(convertObjectToAttributeList(input as any)).toMatchSnapshot()
    });
});