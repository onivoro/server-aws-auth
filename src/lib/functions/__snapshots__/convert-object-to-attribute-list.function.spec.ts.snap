// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertObjectToAttributeList formats object to array, given {"email":null,"address":"","number":0,"age":337} 1`] = `
[
  {
    "Name": "email",
    "Value": "",
  },
  {
    "Name": "phone",
    "Value": "",
  },
  {
    "Name": "address",
    "Value": "",
  },
  {
    "Name": "number",
    "Value": "0",
  },
  {
    "Name": "age",
    "Value": "337",
  },
]
`;
